map/traffic_manager.cpp

```diff
@@ -344,8 +353,10 @@ void TrafficManager::CheckCacheSize()
       auto const it = m_mwmCache.find(mwmId);
       if (it->second.m_isLoaded)
       {
-        m_currentCacheSizeBytes -= it->second.m_dataSize;
+        // Note. It's necessary to multiply by two because routing and rendering use separate caches.
+        m_currentCacheSizeBytes -= 2 * it->second.m_dataSize;
```

Comment:
So, imagine that info.GetColoring.size() is 1. Therefore, 2 * 1 * kElementSize will be assigned to dataSize, so, 2 * kElementSize. This value will be stored in mwmCache, in the field m_dataSize, and m_currentCacheSizeBytes will be increased by 2 * kElementSize.

Imagine now that this entry is evicted from the cache. m_dataSize is 2 * kElementSize, so, m_currentCacheSizeBytes will be decreased by 2 * 2 * kElementSize.

Am I right? If so, your calculations are flawed.